src/classes/CommandParser.ts

```diff
@@ -3,11 +3,9 @@ import { UnknownDictionaryKnownValues } from '../types/common';
 import { parseJSON } from '../lib/helpers';
 
 export default class CommandParser {
-    static getCommand(message: string): string | null {
-        if (message.startsWith('!')) {
-            const index = message.indexOf(' ');
-
-            return message.substring(1, index === -1 ? undefined : index);
+    static getCommand(message: string, prefix: string): string | null {
+        if (message.startsWith(prefix)) {
+            return message.slice(prefix.length).trim().split(/ +/g).shift()?.toLowerCase();
```

Comment:
I actually wondering what's that `split(/ +/g)` for. Could you explain it?

Reply:
If the command goes like this: `!help 1231323` it will split the space between and shift it to only get `help` and not any other arguments for the command. If that explains it.

Reply:
I've used this on my bot without any issues, so don't think so, could make a test case for it to try?

Reply:
Ah, I see. Forgot that the ` +` is a thing in regex.
I guess I will make it split the first space (no need for a global tag).

Thanks for the explanation.